### PR TITLE
added mass matrix regularziation in TSID joint dynamic task

### PR DESCRIPTION
**Pull request by CarlottaSartore**
C.C. @LoreMoretti @Gio-DS

### CODE REVIEW

**Comment by CarlottaSartore** [comment]
@GiulioRomualdi  is still a WIP, I wanted to check the CI, I will ping you when actually ready!

**Comment by S-Dafarra** [comment]
> @GiulioRomualdi is still a WIP, I wanted to check the CI, I will ping you when actually ready!

You can mark it as draft :wink:

**Comment by CarlottaSartore** [comment]
> > @GiulioRomualdi is still a WIP, I wanted to check the CI, I will ping you when actually ready!
> 
> You can mark it as draft 😉

Thanks done ! I was not sure the CI would be triggered by a draft but I think it is the case!

**Comment by CarlottaSartore** [comment]
should be ready now :)

**Review by GiulioRomualdi — COMMENTED** [review]
Hi @CarlottaSartorem thank you for the the PR.  Can you remove the white spaces?
```
The following lines contain trailing whitespace: 

src/TSID/include/BipedalLocomotion/TSID/JointDynamicsTask.h:111:   
src/TSID/src/JointDynamicsTask.cpp:246:    
```

File: src/TSID/include/BipedalLocomotion/TSID/JointDynamicsTask.h [hunk 1]
```diff
@@ -46,7 +46,8 @@ class JointDynamicsTask : public TSIDLinearTask
     System::VariablesHandler::VariableDescription m_jointsTorqueVariable; /**< Variable
                                                                              describing the joint
                                                                              torques */
-
+    bool m_useMassMatrixRegularizationTerm{false};
+    Eigen::MatrixXd m_massMatrixReglarizationTerm;
```
[the remark this answers is not in the document]
Comment (CarlottaSartore) [reply src/TSID/include/BipedalLocomotion/TSID/JointDynamicsTask.h]: Done in dd7832b707262f5531aa0a2860e4ccea991e1406

File: src/TSID/src/JointDynamicsTask.cpp [hunk 2]
```diff
@@ -211,8 +243,18 @@ bool JointDynamicsTask::update()
...
+    
+    if (m_useMassMatrixRegularizationTerm)
+        {
+            iDynTree::toEigen(this->subA(m_robotAccelerationVariable))
+                    = -(m_massMatrix+m_massMatrixReglarizationTerm).bottomRows(m_kinDyn->getNrOfDegreesOfFreedom());
+        } else
+        {
+               iDynTree::toEigen(this->subA(m_robotAccelerationVariable))
         = -m_massMatrix.bottomRows(m_kinDyn->getNrOfDegreesOfFreedom());
+        }
```
[the remark this answers is not in the document]
Comment (CarlottaSartore) [reply src/TSID/src/JointDynamicsTask.cpp]: Done in dd7832b707262f5531aa0a2860e4ccea991e1406

File: src/TSID/src/JointDynamicsTask.cpp [hunk 3]
```diff
@@ -211,8 +243,18 @@ bool JointDynamicsTask::update()
     }
 
     m_b = m_generalizedBiasForces.tail(m_kinDyn->getNrOfDegreesOfFreedom());
-    iDynTree::toEigen(this->subA(m_robotAccelerationVariable))
+    
+    if (m_useMassMatrixRegularizationTerm)
+        {
+            iDynTree::toEigen(this->subA(m_robotAccelerationVariable))
+                    = -(m_massMatrix+m_massMatrixReglarizationTerm).bottomRows(m_kinDyn->getNrOfDegreesOfFreedom());
```
[the remark this answers is not in the document]
Comment (CarlottaSartore) [reply src/TSID/src/JointDynamicsTask.cpp]: Done in dd7832b707262f5531aa0a2860e4ccea991e1406

**Comment by GiulioRomualdi** [comment]
Can you remove the whitespaces in:

```
 The following lines contain trailing whitespace: 

src/TSID/include/BipedalLocomotion/TSID/JointDynamicsTask.h:49:    bool m_useMassMatrixRegularizationTerm{false}; /**< True if the mass matrix regularization is used*/ 
src/TSID/include/BipedalLocomotion/TSID/JointDynamicsTask.h:111:   
src/TSID/src/JointDynamicsTask.cpp:246:    
```

File: src/TSID/src/JointDynamicsTask.cpp [hunk 4]
```diff
@@ -211,8 +243,18 @@ bool JointDynamicsTask::update()
     }
 
     m_b = m_generalizedBiasForces.tail(m_kinDyn->getNrOfDegreesOfFreedom());
-    iDynTree::toEigen(this->subA(m_robotAccelerationVariable))
-        = -m_massMatrix.bottomRows(m_kinDyn->getNrOfDegreesOfFreedom());
+
+    if (m_useMassMatrixRegularizationTerm)
+        {
```
[the remark this answers is not in the document]
Comment (CarlottaSartore) [reply src/TSID/src/JointDynamicsTask.cpp]: Done 2e88facfa5e220a3c6ec66a481d9baaeee95b1f3

**Review by GiulioRomualdi — APPROVED** [review]

**Pull request merged**